### PR TITLE
Use ES6 class style for creating react components

### PR DESCRIPTION
**Pull request by Holi0317**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,3 +14,4 @@ lib
 
 /demos/**/all.*
 /build/
+.idea/
```
Comment (MichaelDeBoey) [inline .gitignore]: This should be in your global `.gitignore` file on your working machine

Comment (Holi0317) [inline .gitignore]: Sorry for forgetting to keep track of this PR ;P

**Comment by MichaelDeBoey** [comment]
@nkbt The `prop-types` is only [compatible](https://github.com/reactjs/prop-types#compatibility) with  `React ^0.14.9` or `React ^15.3.0`

**Comment by nkbt** [comment]
Ouch. Thanks for that @MichaelDeBoey

**Comment by MichaelDeBoey** [comment]
@nkbt Will be fixed with #458

**Comment by nkbt** [comment]
I will publish `0.5.0` (sort of "major" in 0.x way) then, after #457 is merged (tomorrow, since I left my laptop in the office)

**Comment by nkbt** [comment]
Just published `0.5.0` 🤞 

Notes
https://github.com/chenglou/react-motion/releases/tag/v0.5.0

**Pull request merged**